# tfs-ruby-3-2-x-openssl-extconf-ruby-export-msys.patch
#
# Target:  ext/openssl/extconf.rb (pass1 (msys))
# Applies: ruby 3.2.x (3.2.4 - 3.2.7)
# Source:  tebako gem lib/tebako/packager/pass1_patch.rb Pass1MSysPatch::OPENSSL_EXTCONF_RB_PATCH
#
# msys: RUBY_EXPORT=1 shall be set for static builds but is missing in the
# openssl extension; push -DRUBY_EXPORT=1 into $defs.
diff --git a/ext/openssl/extconf.rb b/ext/openssl/extconf.rb
index bc3e4d3..e0607a7 100644
--- a/ext/openssl/extconf.rb
+++ b/ext/openssl/extconf.rb
@@ -189,8 +189,13 @@ have_func("EVP_MD_CTX_get_pkey_ctx(NULL)", evp_h)
 have_func("EVP_PKEY_eq(NULL, NULL)", evp_h)
 have_func("EVP_PKEY_dup(NULL)", evp_h)
 
+# Start of tebako patch
+$defs.push("-DRUBY_EXPORT=1")
+# End of tebako patch
+
 Logging::message "=== Checking done. ===\n"
 
+
 create_header
 create_makefile("openssl")
 Logging::message "Done.\n"
